meta-client/src/main/java/org/apache/helix/metaclient/api/Op.java

```diff
@@ -23,16 +23,96 @@
  *  Represents a single operation in a multi-operation transaction.  Each operation can be a create, set,
  *  version check or delete operation.
  */
-public class Op {
+public abstract class Op {
+  private int type;
+  private String path;
+
+  private Op(int type, String path) {
+    this.type = type;
+    this.path = path;
+  }
+  public static Op create(String path, byte[] data) {
+    return new Create(path, data);
```

Comment:
A little bit confused there. Is this circular call? Op is the abstract class try to create object for Create, which is super class for Op.

Could you please give more explanation here?

Better we have java doc for those APIs.

Reply:
There might be a slight circular call though it is not cyclic. Op creates a Create object which calls its super's (Op) constructor to set the type and path. The implementation of the Op (and OpResult) objects are based on zkclient's implementation of such. It'll be easier for users to adopt metaclient if api resembles that of zkclient.